lib/OpenQA/Git/MaintenanceOutage.pm

```diff
@@ -0,0 +1,37 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::Git::MaintenanceOutage;
+use Mojo::Base -strict, -signatures;
+use Mojo::File 'path';
+
+# Remove or rename these "constants" in favor of a dynamic path
+use constant MAX_DURATION_SECONDS => 1800;
+
+use constant {
+    SKIP => 'SKIP',
+    FAIL => 'FAIL',
+};
+
+sub _state_file ($app) {
+    path($ENV{OPENQA_GIT_SERVER_MAINTENANCE_FILE} // ($app->config->{global}->{cache_dir} // '/var/lib/openqa/cache')
+          . ($ENV{OPENQA_GIT_SERVER_MAINTENANCE_FILE} ? '' : '/gitlab_maintenance.lock'));
```

Comment:
I'd avoid the repetition of the lengthy `$ENV{OPENQA_GIT_SERVER_MAINTENANCE_FILE}`.

Reply:
changed it to `path($ENV{OPENQA_GIT_SERVER_MAINTENANCE_FILE} // ($app->config->{global}->{cache_dir} // '/var/lib/openqa/cache') . '/gitlab_maintenance.lock');`
I wanted to make sure that '/gitlab_maintenance.lock' is not appended if a path was provided via a env variable, but i guess this still works due to operator precedence.. ?